satpy/readers/li_l2_nc.py

```diff
@@ -158,11 +158,9 @@ def get_array_on_fci_grid(self, data_array: xr.DataArray):
         data_2d = da.map_blocks(_np_add_at_wrapper, data_2d, (rows, cols), data_array,
                                 dtype=data_array.dtype,
                                 chunks=(LI_GRID_SHAPE[0], LI_GRID_SHAPE[1]))
-        data_2d = da.where(data_2d > 0, data_2d, np.nan)
-
+        data_2d = da.where(data_2d > 0, data_2d, np.float32(np.nan)).astype(np.float32)

```

Comment:
does it make a difference to have the  `np.float32(np.nan)` here or is it enough with the `astype`?

Reply:
No it does not make difference I have changed `np.float32(np.nan)` --> `np.nan`